android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -459,12 +459,13 @@ constructor(
 
   private suspend fun updateCarePlanAndTask(extras: List<Resource>, backReference: String?) {
     if (backReference != null && backReference != "notify") {
-      extras.forEach {
-        if (it is Encounter) {
-          fhirCarePlanGenerator.completeTask(
-            backReference.asReference(ResourceType.Task).extractId(),
-            it.status,
-          )
+      extras.forEach { resource ->
+        if (resource is Encounter) {
+          val taskReference = backReference.asReference(ResourceType.Task)
+          val taskID = taskReference.extractId()
+          val encounterStatus = resource.status
+          val encounterReference = resource.logicalId.removePrefix("#")
```

Comment:
This should be converted into a reference, you can call asReference like how we did with the task